rfcs/rfc-net-20230821-1-headless-server-launcher.md

```diff
@@ -0,0 +1,128 @@
+### Summary:
+This RFC is to propose support for O3DE headless server launchers. Currently the server launcher is launched 'headlessly' by specifying the Atom null-renderer as either part of the command line or specified in a custom `.setreg` or `.cfg` file for the game project. When headless server mode is activated this way, the server launcher will simply suppress the creation of the client window and proceed with the server launcher process. However, to be a truly headless server, the determination of a graphical client vs a console application needs to be determined at compile time when the server launcher to built. While the behavior varies by platform, CMake requires that an executable be identified by the target type of either `APPLICATION` or `EXECUTABLE`. To make the server launcher truly headless, updates will be needed to refactor how the server launcher and related libraries are defined.
+
+### What is the relevance of this feature?
+The general use case for dedicated game servers do not need any graphical client since there is no user interaction necessary. Server applications normally just need to launch and shutdown. Launching is done through executing the application, and shutdown can be done through a terminate signal (CTRL-C) or a kill request. Being able to build a true headless server launcher will remove the requirement of having a GUI manager installed on the machine that will host the server.
+
+### Feature design description:
+To maintain backwards compatibility, an additional `HeadlessServerLauncher` target will be built in addition to the existing `ServerLauncher` target. The `HeadlessServerLauncher`  will behave in the follow ways:
+
+1.  On Windows, the server launcher will not spawn any native window and return immediately. Instead, it will behave like a true console application (It will be linked with `/SUBSYSTEM:CONSOLE` instead of `/SUBSYSTEM:WINDOWS`). It will direct all of its output to the current terminal / command window from where it is executed. If it double-clicked from a file explorer, it will behave as with any console application: A default windows terminal will launch displaying the console stdout/stderr of the server process.
```

Comment:
```suggestion
1.  On Windows, the server launcher will not spawn any native window and return immediately. Instead, it will behave like a true console application (It will be linked with `/SUBSYSTEM:CONSOLE` instead of `/SUBSYSTEM:WINDOWS`). It will direct all of its output to the current terminal / command window from where it is executed. If double-clicked from a file explorer, it will behave as a console application; a default Windows terminal will launch displaying the console stdout/stderr of the server process.
```